### DISCUSSION TITLE
Only support "sha-256"? What about "SHA-256"

### DISCUSSION THREAD

**Discussion by DrakF12** [paullouisageneau/libdatachannel#948] [General]
```c++
if (match_prefix(value, "sha-256 ")) {
    string fingerprint{value.substr(8)};
    trim_begin(fingerprint);
    setFingerprint(std::move(fingerprint));
} else {
    PLOG_WARNING << "Unknown SDP fingerprint format: " << value;
}
``` 
I have tested OBS studio's ability to support WHIP, which use libdatachannel to achieve the rtc connection.
But I find the SDP only supports "sha-256", not include "SHA-256".
Is "sha-256" normalized? Or is "SHA-256' not normalized?

**Comment by paullouisageneau** [comment]
According to [RFC 4572 section 5](https://www.rfc-editor.org/rfc/rfc4572#section-5) the hash function name must be lowercase:
```
hash-func              =  "sha-1" / "sha-224" / "sha-256" /
                             "sha-384" / "sha-512" /
                             "md5" / "md2" / token
                             ; Additional hash functions can only come
                             ; from updates to [RFC 3279](https://www.rfc-editor.org/rfc/rfc3279)
```

However matching could be made case-insensitive if it helps interoperability.

**Reply by DrakF12** [reply]
Thanks, I think I should change "SHA-256" into "sha-256".
